### PR TITLE
WIP: Serde Logic for PointerTensor

### PR DESCRIPTION
**Pull request by Nivek92**

### CODE REVIEW

File: syft/workers/base.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def request_obj(self, obj_id, location):
 
     # SECTION: Manage the workers network
 
-    def get_worker(self, id_or_worker):
+    def get_worker(self, id_or_worker, fail_hard=False):
```
Comment (Ogofo) [inline syft/workers/base.py]: Documentation for `fail_hard` argument is missing.

File: test/torch/tensors/test_tensor.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+from unittest import TestCase
+import torch
+
+
+import syft
+from syft.frameworks.torch.tensors import TorchTensor, PointerTensor
+
+
+class TestNative(TestCase):
```
Comment (Ogofo) [inline test/torch/tensors/test_tensor.py]: I think we switched `TestCase` to object. Pytest does not require `TestCase`, but `TestCase` is disallowing the use of parametrizing methods.

File: test/torch/tensors/test_tensor.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    def test_init(self):
+        tensor_extension = TorchTensor()
+        assert tensor_extension.id is None
+        assert tensor_extension.owner is None
+
+
+class TestPointer(TestCase):
```
Comment (Ogofo) [inline test/torch/tensors/test_tensor.py]: `TestCase`: see above

File: syft/serde.py [hunk 4]
```diff
@@ -40,6 +40,10 @@
 import io
 import numpy
 import zstd
+import syft
```
Comment (Ogofo) [inline syft/serde.py]: Is this import really needed?

**Comment by Nivek92** [comment]
I noticed, that in this version you have to register all workers to "me" aka the local_worker otherwise we do not get a virtual worker during deserialization but a string with the id of the worker (e.g. Pointer.location will be "Alice" instead of the VirtualWorker with id Alice) 

I am not sure if that is the desired/expected behavior.

File: syft/serde.py (hunk 4 above)
Comment (Nivek92) [reply syft/serde.py]: Yeah because I want the local_worker and thus call `syft.local_worker.get_worker(val_str, fail_hard=True)`

**Comment by iamtrask** [comment]
I attempted a merge for you which is now in this branch https://github.com/OpenMined/PySyft/tree/Nivek92-fallback

**Pull request merged**